src/serviceClients/sqliteClient.ts

```diff
@@ -85,6 +85,44 @@ export class SQLiteClient {
     }
   }
 
+  public getGpkgTileWidthAndHeight(): { tileWidth: number; tileHeight: number } {
+    let db: SQLiteDB | undefined = undefined;
+    try {
+      db = new Database(this.fullPath, { fileMustExist: true });
```

Comment:
if the line `db = new Database(this.fullPath, { fileMustExist: true });` fails please raise BadRequestError with invalid GPKG file, for other parts of this method keep the current error handling

Reply:
please add this logic to other methods in sqliteClient that create a new Database()